src/watertap_contrib/seto/unit_models/surrogate/vagmd_surrogate.py

```diff
@@ -0,0 +1,1233 @@
+            )
+
+        @self.Constraint(doc="Permeate flow rate")
+        def eq_permeate_volumetric_flow_rate(b):
+            return b.permeate_props[0].flow_vol_phase["Liq"] == pyunits.convert(
+                b.permeate_flux * self.module_area,
+                to_units=pyunits.m**3 / pyunits.s,
+            )
+
+        @self.Constraint(doc="Condenser mass flow rate")
+        def eq_condenser_mass_flow_rate(b):
+            return (
+                b.avg_condenser_props[0].flow_mass_phase_comp["Liq", "H2O"]
+                == b.permeate_props[0].flow_mass_phase_comp["Liq", "H2O"]
+            )
+
+        @self.Constraint(doc="initial log mean temperature difference")
+        def eq_log_mean_temp_dif(b):
+            return b.log_mean_temp_dif == ((TEI - TCO) - (TEO - TCI)) / log(
+                (TEI - TCO) / (TEO - TCI + 1e-6)
+            )
+
+        @self.Constraint(doc="Thermal power")
+        def eq_thermal_power(b):
+            CpF = b.avg_feed_props[0].cp_mass_phase["Liq"]
+            RhoF = b.avg_feed_props[0].dens_mass_phase["Liq"]
+            return b.thermal_power == pyunits.convert(
+                (FFR * CpF * (TEI - TCO)) * (RhoF), to_units=pyunits.kW
+            )
+
+        @self.Constraint(doc="Electric power for pumping feed water")
+        def eq_feed_pump_power_elec(b):
+            return b.feed_pump_power_elec == pyunits.convert(
+                b.feed_flow_pressure_drop
+                / b.pump_efficiency
+                * b.feed_props[0].flow_vol_phase["Liq"],
+                to_units=pyunits.kW,
+            )
+
+        @self.Constraint(doc="Electric power for pumping cooling water")
+        def eq_cooling_pump_power_elec(b):
+            return b.cooling_pump_power_elec == pyunits.convert(
+                b.cooling_flow_pressure_drop
+                / b.pump_efficiency
+                * b.cooling_in_props[0].flow_vol_phase["Liq"],
+                to_units=pyunits.kW,
+            )
+
+        @self.Constraint(doc="Calculate hot side thermal resistance")
+        def eq_thermal_resistance_hot(b):
+            return b.thermal_resistance_hot == pyunits.convert(
+                b.feed_props[0].flow_vol_phase["Liq"]
+                * b.avg_feed_props[0].dens_mass_phase["Liq"]
+                * b.avg_feed_props[0].cp_mass_phase["Liq"],
+                to_units=pyunits.W / pyunits.K,
+            )
+
+        @self.Constraint(doc="Calculate cold side thermal resistance")
+        def eq_thermal_resistance_cold(b):
+            return b.thermal_resistance_cold == pyunits.convert(
+                b.cooling_in_props[0].flow_vol_phase["Liq"]
+                * b.avg_cooling_props[0].dens_mass_phase["Liq"]
+                * b.avg_cooling_props[0].cp_mass_phase["Liq"],
+                to_units=pyunits.W / pyunits.K,
+            )
+
+        @self.Constraint(doc="Calculate the number of transfer units")
+        def eq_number_transfer_units(b):
```

Comment:
I assume this doesn't solve to an integer. Any issues with that or how do you consider that? Or maybe its ok if it is a fraction?

Reply:
Yes. It's just a thermodynamic coefficient, but with a confusing name. It is okay to be a fraction.